WordPress/src/test/java/org/wordpress/android/ui/blaze/blazecampaigns/CampaignListingViewModelTest.kt

```diff
@@ -0,0 +1,204 @@
+    fun `given no campaigns, when click is invoked on create, then navigate to blaze flow`() = runTest {
+        val siteModel = mock<SiteModel>()
+        val noCampaigns = BlazeCampaignsModel(emptyList(), 1, 1, 0)
+        whenever(networkUtilsWrapper.isNetworkAvailable()).thenReturn(true)
+        whenever(selectedSiteRepository.getSelectedSite()).thenReturn(siteModel)
+        whenever(blazeCampaignsStore.getBlazeCampaigns(siteModel)).thenReturn(noCampaigns)
+
+        viewModel.start(CampaignListingPageSource.DASHBOARD_CARD)
+        val noCampaignsState = uiStates.last() as CampaignListingUiState.Error
+        noCampaignsState.button!!.click.invoke()
+
+        assertThat(navigationEvents.last() is CampaignListingNavigation.CampaignCreatePage).isTrue
+    }
+
+    @Test
+    fun `given campaigns available, when clicked on campaign, then navigate to detail`() = runTest {
+        val siteModel = mock<SiteModel>()
+        setupCampaignModels()
+        whenever(networkUtilsWrapper.isNetworkAvailable()).thenReturn(true)
+        whenever(selectedSiteRepository.getSelectedSite()).thenReturn(siteModel)
+        whenever(blazeCampaignsStore.getBlazeCampaigns(siteModel)).thenReturn(campaignModel)
+        whenever(statsUtils.toFormattedString(1000L, ONE_THOUSAND)).thenReturn("1000")
+
+
+        viewModel.start(CampaignListingPageSource.DASHBOARD_CARD)
+        val success = uiStates.last() as CampaignListingUiState.Success
+        success.itemClick.invoke(success.campaigns.first())
+
+        assertThat(navigationEvents.last() is CampaignListingNavigation.CampaignDetailPage).isTrue
+    }
+
+    @Test
+    fun `given campaigns available, when clicked on create campaign fab, then navigate to blaze flow`() = runTest {
+        val siteModel = mock<SiteModel>()
+        setupCampaignModels()
+        whenever(networkUtilsWrapper.isNetworkAvailable()).thenReturn(true)
+        whenever(selectedSiteRepository.getSelectedSite()).thenReturn(siteModel)
+        whenever(blazeCampaignsStore.getBlazeCampaigns(siteModel)).thenReturn(campaignModel)
+        whenever(statsUtils.toFormattedString(1000L, ONE_THOUSAND)).thenReturn("1000")
+
+
+        viewModel.start(CampaignListingPageSource.DASHBOARD_CARD)
+        val success = uiStates.last() as CampaignListingUiState.Success
+        success.createCampaignClick.invoke()
+
+        assertThat(navigationEvents.last() is CampaignListingNavigation.CampaignCreatePage).isTrue
+    }
+
+    private fun setupCampaignModels() {
+        campaignModel = BlazeCampaignsModel(
+            campaigns = getCampaigns(10),
+            page = 1,
+            totalPages = 1,
+            totalItems = 10
+        )
+    }
+
+    private fun getCampaigns(numbers: Int): List<BlazeCampaignModel> {
+        val listOfCampaigns = mutableListOf<BlazeCampaignModel>()
+        for (i in 0 until numbers) {
+            listOfCampaigns.add(
+                BlazeCampaignModel(
+                    campaignId = i.toLong(),
```

Comment:
@AjeshRPai - This doesn't need the `toLong` applied and it is causing a compile error. I have removed it and pushed a commit.

Reply:
Thank you